pkg/services/accesscontrol/roles.go

```diff
@@ -195,28 +169,65 @@ const (
 
 	ldapAdminEdit = "fixed:ldap:admin:edit"
 	ldapAdminRead = "fixed:ldap:admin:read"
+)
 
-	provisioningAdmin = "fixed:provisioning:admin"
+var (
+	once sync.Once
+
+	// FixedRoles provides a map of permission sets/roles which can be
+	// assigned to a set of users. When adding a new resource protected by
+	// Grafana access control the default permissions should be added to a
+	// new fixed role in this set so that users can access the new
+	// resource. FixedRoleGrants lists which built-in roles are
+	// assigned which fixed roles in this list.
+	FixedRoles      = map[string]RoleDTO{}
```

Comment:
Might be like
```go
type FixedRolesMap struct {
	roles map[string]RoleDTO{}
	mutex sync.RWMutex
}

func (r *FixedRolesMap) Load(name string) (RoleDTO, bool) {
	r.mutex.RLock()
	defer r.mutex.RUnlock()
	role, exists := r.roles[name]
	return role, exists
}

func (r *FixedRolesMap) Store(name string, role RoleDTO) {
	r.mutex.Lock()
	defer r.mutex.Unlock()
	r.roles[name] = role
	return
}

var FixedRoles = FixedRolesMap{}
```

Reply:
Doing exactly that after our discussion and suggestion by @joanlopez :+1:

Reply:
Yeah, that's a common pattern

Reply:
I guess `sync.Map` is very similar internally, but Go has no generics yet, so in order to build a common type, they should use `interface{}` and loose type safety.